commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddLdapAuthentication.java

```diff
@@ -0,0 +1,192 @@
+package org.wildfly.extras.creaper.commands.security.realms;
+
+import java.io.IOException;
+import org.wildfly.extras.creaper.commands.foundation.offline.xml.GroovyXmlTransform;
+import org.wildfly.extras.creaper.commands.foundation.offline.xml.Subtree;
+import org.wildfly.extras.creaper.core.ManagementVersion;
+import org.wildfly.extras.creaper.core.offline.OfflineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+/**
+ * Add ldap authentication to security realm.
+ */
+public final class AddLdapAuthentication extends AbstractAddSecurityRealmSubElement {
+
+    private final String advancedFilter;
+    private final Boolean allowEmptyPasswords;
+    private final String baseDn;
+    private final String connection;
+    private final Boolean recursive;
+    private final String userDn;
+    private final String usernameAttribute;
+    private final String usernameLoad;
+    private final LdapCache cache;
+
+    private AddLdapAuthentication(Builder builder) {
+        super(builder);
+        this.advancedFilter = builder.advancedFilter;
+        this.allowEmptyPasswords = builder.allowEmptyPasswords;
+        this.baseDn = builder.baseDn;
+        this.connection = builder.connection;
+        this.recursive = builder.recursive;
+        this.userDn = builder.userDn;
+        this.usernameAttribute = builder.usernameAttribute;
+        this.usernameLoad = builder.usernameLoad;
+        this.cache = builder.cache;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws Exception {
+        if (usernameLoad != null) {
+            ctx.client.serverVersion().assertAtLeast(ManagementVersion.VERSION_2_0_0,
```

Comment:
The version is available directly on the `OnlineCommandContext`: `ctx.serverVersion`